drivers/video/video_stm32_dcmi.c

```diff
@@ -0,0 +1,511 @@
+	if (ret != 0) {
+		LOG_ERR("Failed to configure DMA channel %d",
+			data->dma.channel + STM32_DMA_STREAM_OFFSET);
+		return ret;
+	}
+
+	/*** Configure the DMA ***/
+	/* Set the parameters to be configured */
+	hdma.Init.Request		= DMA_REQUEST_DCMI;
+	hdma.Init.Direction		= DMA_PERIPH_TO_MEMORY;
+	hdma.Init.PeriphInc		= DMA_PINC_DISABLE;
+	hdma.Init.MemInc		= DMA_MINC_ENABLE;
+	hdma.Init.PeriphDataAlignment	= DMA_PDATAALIGN_WORD;
+	hdma.Init.MemDataAlignment	= DMA_MDATAALIGN_WORD;
+	hdma.Init.Mode			= DMA_CIRCULAR;
+	hdma.Init.Priority		= DMA_PRIORITY_HIGH;
+	hdma.Init.FIFOMode		= DMA_FIFOMODE_DISABLE;
+
+	hdma.Instance = __LL_DMA_GET_STREAM_INSTANCE(data->dma.reg,
+						data->dma.channel);
+
+	/* Initialize DMA HAL */
+	__HAL_LINKDMA(&data->hdcmi, DMA_Handle, hdma);
+
+	if (HAL_DMA_Init(&hdma) != HAL_OK) {
+		LOG_ERR("DCMI DMA Init failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int stm32_dcmi_enable_clock(const struct device *dev)
+{
+	const struct video_stm32_dcmi_config *config = dev->config;
+	const struct device *dcmi_clock = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+	int err;
+
+	if (!device_is_ready(dcmi_clock)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	/* Turn on DCMI peripheral clock */
+	err = clock_control_on(dcmi_clock, (clock_control_subsys_t *) &config->pclken);
+	if (err < 0) {
+		LOG_ERR("Failed to enable DCMI clock");
+		LOG_ERR("Error code %d: %s", err, strerror(-err));
+		return err;
+	}
+
+	return 0;
+}
+
+static int video_stm32_dcmi_set_fmt(const struct device *dev,
+				  enum video_endpoint_id ep,
+				  struct video_format *fmt)
+{
+	const struct video_stm32_dcmi_config *config = dev->config;
+	struct video_stm32_dcmi_data *data = dev->data;
+	unsigned int bpp = video_pix_fmt_bpp(fmt->pixelformat);
+
+	if (!bpp || ep != VIDEO_EP_OUT) {
+		return -EINVAL;
+	}
+
+	data->pixel_format = fmt->pixelformat;
+	data->pitch = fmt->pitch;
+	data->height = fmt->height;
+	data->width = fmt->width;
+
+	if (video_set_format(config->sensor_dev, ep, fmt)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int video_stm32_dcmi_get_fmt(const struct device *dev,
+				  enum video_endpoint_id ep,
+				  struct video_format *fmt)
+{
+	struct video_stm32_dcmi_data *data = dev->data;
+	const struct video_stm32_dcmi_config *config = dev->config;
+
+	if ((fmt == NULL) || (ep != VIDEO_EP_OUT)) {
+		return -EINVAL;
+	}
+
+	if (!video_get_format(config->sensor_dev, ep, fmt)) {
+		/* align DCMI with sensor fmt */
+		return video_stm32_dcmi_set_fmt(dev, ep, fmt);
+	}
+
+	fmt->pixelformat = data->pixel_format;
+	fmt->height = data->height;
+	fmt->width = data->width;
+	fmt->pitch = data->pitch;
+
+	return 0;
+}
+
+static int video_stm32_dcmi_stream_start(const struct device *dev)
+{
+	struct video_stm32_dcmi_data *data = dev->data;
+	const struct video_stm32_dcmi_config *config = dev->config;
+	size_t buffer_size = data->pitch * data->height;
+
+	LOG_INF("Start stream capture");
```

Comment:
sure you want these as `LOG_INF` the final driver? normally I'd expect these to be `LOG_DBG`

Reply:
Removed